kedro/pipeline/pipeline.py

```diff
@@ -380,6 +381,21 @@ def only_nodes(self, *node_names: str) -> "Pipeline":
         """
         unregistered_nodes = set(node_names) - set(self._nodes_by_name.keys())
         if unregistered_nodes:
+            # check if unregistered nodes are available under namespace
+            namespaces = []
+            for name in node_names:
```

Comment:
> 1. Elsewhere in kedro we use `difflib`, e.g. to [suggest catalog entry names](https://github.com/kedro-org/kedro/blob/main/kedro/io/data_catalog.py#L292) or CLI commands that are mistyped. Did you consider using that here rather than coding our own very simple matching algorithm? I don't know what the capabilities of `difflib` are so maybe it doesn't have anything suitable for this case, but worth checking.

`difflib` does offer some interesting capabilities, especially if we go for the route suggested in the second point, but I don't think the nature of its implementation works in this context. A quick test reveals the `get_close_matches` will omit the "correct" suggestions in favour for unrelated nodes that it computes as being more similar to the inputs 😬.

Reply:
> Surely we only need to check through unregistered_nodes?

Thanks for catching that! As a consequence the error message will no longer display the incorrect nodes in the order they were provided by the user - is this behaviour undesirable or should it be ignored?

Reply:
I think it's fine - the order isn't important here.